packages/ui/src/components/Visualization/Canvas/CanvasForm.tsx

```diff
@@ -122,7 +123,14 @@ export const CanvasForm: FunctionComponent<CanvasFormProps> = (props) => {
     return schema?.schema && schema.schema['$comment']?.includes('loadbalance');
   }, [schema]);
 
-  return schema?.schema === undefined ? null : (
+  return schema?.schema === undefined || Object.keys(schema?.schema).length === 0 ? (
+    <>
+      <Title headingLevel="h1">{componentName}</Title>
+      <CodeBlock>
+        <CodeBlockCode id="code-content">{stringify(model)}</CodeBlockCode>
+      </CodeBlock>
+    </>
```

Comment:
:nail_care: Maybe it would be nice to extract this into a dedicated component.

Reply:
It could look like this
```tsx
<UnknownComponent componentName={componentName} } code={stringify(model)} />
```

Alternatively, we could reorder a bit the component to avoid duplicating the `<Title>` component:
```tsx

  const isUnknownComponent = useMemo(() => {
    return schema?.schema === undefined || Object.keys(schema?.schema).length === 0;
  }, [schema]);

  return (
    <ErrorBoundary key={props.selectedNode.id} fallback={<p>This node cannot be configured yet</p>}>
      <Title headingLevel="h1">{componentName}</Title>
      {isUnknownComponent ? (
        <CodeBlock>
          <CodeBlockCode>{stringify(props.selectedNode.data?.vizNode?.getComponentSchema())}</CodeBlockCode>
        </CodeBlock>
      ) : (
        <AutoField.componentDetectorContext.Provider value={CustomAutoFieldDetector}>
          {isExpressionAwareStep && <StepExpressionEditor selectedNode={props.selectedNode} />}
          {isDataFormatAwareStep && <DataFormatEditor selectedNode={props.selectedNode} />}
          {isLoadBalanceAwareStep && <LoadBalancerEditor selectedNode={props.selectedNode} />}
          <AutoForm ref={formRef} schema={schema} model={model} onChangeModel={handleOnChange} data-testid="autoform">
            <AutoFields omitFields={omitFields} />
            <ErrorsField />
          </AutoForm>
        </AutoField.componentDetectorContext.Provider>
      )}
    </ErrorBoundary>
  );
```

Reply:
Thanks for the hint Ricardo, I updated the PR to remove the duplicate `<Title>` and also updated the failing tests.